src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -411,4 +411,17 @@ def word_break(string, length)
     # adds a <wbr> tag after an amount of given characters
     safe_join(string.scan(/.{1,#{length}}/), "<wbr>".html_safe)
   end
+
+  def toggle_sliced_text(text, slice_length = 50, id = "toggle_sliced_text_#{Time.now.to_f.to_s.delete('.')}")
+    return text if text.length < slice_length
+    short = content_tag(:span, 'data-toggle-id' => id) do
+      content_tag(:span, text.slice(0, slice_length) + ' ') +
+      link_to('[+]', '#', onclick: "$(\"[data-toggle-id='#{id}']\").toggle();".html_safe)
```

Comment:
Yes, but if some developer uses your function wrong you have an XSS, like

``
def toggle_sliced_text(text, 50, "toggle_#{user.name}_description") 
``

you probably have to sth like:

```
js = '$(\"[data-toggle-id='.html_safe
js << id
js << '").toggle();'.html_safe
```
With my previous comment you would need to do the html_safe on the whole string again!